### PR TITLE
[Bugfix] Exclude DP token padding from draft attention metadata

### PR DESCRIPTION
**Pull request by khluu**
DeepSeek-V3.2 DP8 + EP + MTP crashes when DP ranks have unequal request counts. A rank with 12 requests receives model inputs padded to 13 tokens; the sparse indexer then fails with `shape '[12, -1, 64, 128]' is invalid for input of size 106496`. This is confirmed in [main build 87584](https://buildkite.com/vllm/ci/builds/87584#01a07dad-00e7-4a29-9abe-6cdc65c9100f). The subsequent GSM8K result of 0.0000 reflects the engine failure.

Autoregressive draft decode produces one query per request. Use the descriptor's request count, falling back to the actual local request count, for the attention token count in both ordinary and fused multi-step decode. Preserve padded model inputs/collective sizes and FULL graph request padding.

Validation:
- The exact [H200 large-model job in build 87389](https://buildkite.com/vllm/ci/builds/87389#01a0712c-2e18-480d-bda7-dfac8725ed44) passed all **6 evaluations** on implementation commit `39ba6cbae8e1`. DeepSeek-V3.2-DP scored **0.9568**, above its **0.8700** lower bound.
- After merging main and removing the added regression test, unused imports caused pre-commit to fail. Those imports are now removed. `.venv/bin/pre-commit run --files tests/v1/spec_decode/test_eagle_draft_attn_metadata.py vllm/v1/worker/gpu/spec_decode/autoregressive/speculator.py` and `git diff --check` passed.
- Fresh targeted H200 validation for current head `c34119ecbe4d` is [build 87771](https://buildkite.com/vllm/ci/builds/87771). Its result is pending; the older passing run above is not represented as current-head validation.

Duplicate checks found no competing PR for the one-query-per-request draft-decode metadata mismatch. Related #53660, #54102, and #53577 address draft prefill, verifier residual padding, and DSpark PP respectively. This PR updates the existing fix rather than introducing a second implementation.

AI assistance was used for investigation, implementation, and validation.

### CODE REVIEW

File: tests/v1/spec_decode/test_eagle_draft_attn_metadata.py [hunk 1]
```diff
@@ -159,3 +166,69 @@ def fake_prepare(out, seq_lens, num_reqs, dcp_size, dcp_rank, cp_interleave):
...
+            skip_attn=False,
+            batch_desc=BatchExecutionDescriptor(
+                cg_mode=cg_mode,
+                num_tokens=padded_tokens,
+                num_reqs=padded_tokens if full else None,
+            ),
+            num_tokens_across_dp=torch.tensor([num_reqs, padded_tokens]),
+            seq_lens_cpu_upper_bound=torch.full((16,), 100, dtype=torch.int32),
+        )
+    assert metadata_batches == [expected_reqs] * (1 if fused else 2)
```
Comment (yewentao256) [inline tests/v1/spec_decode/test_eagle_draft_attn_metadata.py]:
```suggestion
```
I don't think we need a specific unit test for this small update

**Review by yewentao256 — COMMENTED** [review]
Thanks for the work!

**Review by yewentao256 — APPROVED** [review]
LGTM, thanks for the work!

**Comment by njhill** [comment]
I'm curious do we know which change introduced/exposed this?

**Comment by TheEpicDolphin** [comment]
I'm a bit late to review this, but the changes LGTM.

I'm following up with a fix for dflash/dspark which are at risk of the same type of issue: https://github.com/vllm-project/vllm/pull/56181
